ros_buildfarm/templates/__init__.py

```diff
@@ -91,26 +98,68 @@ def _expand_template(template_name, **kwargs):
             sys.exit(1)
 
 
-def create_dockerfile(template_name, data, dockerfile_dir):
-    data['template_name'] = template_name
+def _find_first_template(template_name, **kwargs):
+    """Return first found template resource respective to template_packages order."""
+    if 'template_packages' not in kwargs:
+        # Default to ros_buildfarm if none given
+        return 'ros_buildfarm'
+
+    # Look through each given template package
+    for template_package in kwargs['template_packages']:
+        if pkg_resources.resource_exists(template_package, template_name):
+            return template_package
+
+    # Default to ros_buildfarm if none found
+    return 'ros_buildfarm'
 
-    wrapper_scripts = {}
-    wrapper_script_path = os.path.join(
-        os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
-        'scripts', 'wrapper')
-    for filename in os.listdir(wrapper_script_path):
-        if not filename.endswith('.py'):
-            continue
-        abs_file_path = os.path.join(wrapper_script_path, filename)
-        with open(abs_file_path, 'r') as h:
-            content = h.read()
-            wrapper_scripts[filename] = content
-    data['wrapper_scripts'] = wrapper_scripts
 
+def _find_first_wrappers(**data):
+    """Return first found wrappers resource respective to template_packages order."""
+    wrapper_scripts = {}
+    wrapper_subpath = 'templates/wrapper'
+    for template_package in data['template_packages']:
+        if pkg_resources.resource_exists(template_package, wrapper_subpath):
+            wrapper_path = pkg_resources.resource_filename(template_package, wrapper_subpath)
+            wrapper_files = pkg_resources.resource_listdir(template_package, wrapper_subpath)
+            for filename in wrapper_files:
+                if not filename.endswith('.py'):
+                    continue
+                if filename in wrapper_scripts:
+                    continue
+                abs_file_path = os.path.join(wrapper_path, filename)
+                with open(abs_file_path, 'r') as h:
+                    content = h.read()
+                    wrapper_scripts[filename] = content
+    return wrapper_scripts
+
+
+def create_dockerfile(data):
+    """Write Dockerfile to disk using data and template."""
+
+    # Create copy of data
+    data = dict(data)
+
+    # Prepend templates folder to name
+    data['template_name'] = os.path.join('templates',  data['template_name'])
+
+    template_name = data['template_name']
+    dockerfile_dir = data['dockerfile_dir']
+
+    # Find first instance of tempate_name given order of template_packages
+    template_package = _find_first_template(**data)
```

Comment:
Passing `template_name` as the first argument and not putting it in the dictionary which is passed with `**`.